google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/spi/v1/FakeScheduledExecutorService.java

```diff
@@ -182,6 +204,13 @@ public void execute(Runnable command) {
       pendingCallables.add(callable);
     }
     work();
+    synchronized (expectedWorkQueue) {
```

Comment:
So we are not waiting here, but I have added documentation that allows you to wait for the work to be scheduled as well as to the method that allows you to set the expectations, I hope that is sufficient.